### DISCUSSION TITLE
Encrypting and Decrypting Using A256GCM Algorithm

### DISCUSSION THREAD

**Discussion by sohlae** [AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet#2607] [Q&A] [answered]
Hi, when I attempted to encrypt a JWT using the A256GCM algorithm, I received this exception. I am using version **6.35.0** of the Microsoft.IdentityModel.JsonWebTokens library.

`Microsoft.IdentityModel.Tokens.SecurityTokenEncryptionFailedException: 'IDX10617: Encryption failed. Keywrap is only supported for: 'A128CBC-HS256', 'A192CBC-HS384' and 'A256CBC-HS512'. The content encryption specified is: 'A256GCM'.'`

I have a few questions:
1. Does this mean that if I try to decrypt a JWE that was encrypted using this algorithm, it will fail as well?
2. Are there alternatives that I can use to be able to use A256GCM in .NET 6.0?

**Comment by brentschmaltz** [comment, marked as answer]
@sohlae We do not support creating JWE's with A256GCM, we do decrypt such tokens.
